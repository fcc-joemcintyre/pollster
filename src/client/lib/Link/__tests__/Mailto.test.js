/* eslint-env jest */
import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import { Mailto, PlainMailto } from '..';

const theme = {
};

test ('Mailto, to only, default render', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <Mailto to='#'>text</Mailto>
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('Mailto, default render', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <Mailto to='#' subject='Hi'>text</Mailto>
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('PlainMailto, to only, default render', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <PlainMailto to='#'>text</PlainMailto>
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('PlainMailto, default render', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <PlainMailto to='#' subject='Hi'>text</PlainMailto>
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('PlainMailto with color, to only, default render', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <PlainMailto to='#' c='#0000ff'>text</PlainMailto>
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('PlainMailto with color, default render', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <PlainMailto to='#' subject='Hi' c='#0000ff'>text</PlainMailto>
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});
