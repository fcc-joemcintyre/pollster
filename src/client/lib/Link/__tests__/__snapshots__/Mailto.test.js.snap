// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mailto, default render 1`] = `
<a
  href="mailto:#?subject=Hi"
>
  text
</a>
`;

exports[`Mailto, to only, default render 1`] = `
<a
  href="mailto:#null"
>
  text
</a>
`;

exports[`PlainMailto with color, default render 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #0000ff;
}

.c0:visited: {
  color: #0000ff;
}

<a
  c="#0000ff"
  className="c0"
  href="mailto:#?subject=Hi"
>
  text
</a>
`;

exports[`PlainMailto with color, to only, default render 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #0000ff;
}

.c0:visited: {
  color: #0000ff;
}

<a
  c="#0000ff"
  className="c0"
  href="mailto:#null"
>
  text
</a>
`;

exports[`PlainMailto, default render 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  className="c0"
  href="mailto:#?subject=Hi"
>
  text
</a>
`;

exports[`PlainMailto, to only, default render 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  className="c0"
  href="mailto:#null"
>
  text
</a>
`;
